### PR TITLE
client_name parameter to allow applications to identify themselves

### PR DESCRIPTION
**Pull request by sanderland**
As requested by @egrefen

### CODE REVIEW

File: cohere/client.py [hunk 1]
```diff
@@ -35,13 +35,28 @@ def __init__(self,
...
+                 client_name: str = None) -> None:
+        """
+        Initialize the client.
+        Args:
+           * api_key (str): Your API key.
+           * version (str): API version to use. Will use cohere.COHERE_VERSION by default.
+           * num_workers (int): Maximal number of threads for parallelized calls.
+           * request_dict (dict): Additional parameters for calls to requests.post
+           * check_api_key (bool): Whether to check the api key for validity on initialization.
+           * client_name (str): A string to identify your application for analytics purposes.
```
Comment (egrefen) [inline cohere/client.py:48]: Should we indicate that it's for analytics purposes _on the cohere end?_ We don't expect to give external access to this data, do we?

Developers who need to do tracking for analytics can do so in the application which encapsulates API calls made via the SDK.

Comment (sanderland) [reply cohere/client.py:48]: I think it does not make a lot of sense to allow it per call. If you really need to (multiple sub-applications?), you could simply instantiate multiple clients.

Comment (egrefen) [reply cohere/client.py:48]: Fair enough.

Comment (egrefen) [reply cohere/client.py:48]: Wait, this is responding to my other comment, no? What about the point about the docstring?

Comment (sanderland) [reply cohere/client.py]: Don't really have a strong opinion on that one, but added 'internal'.

**Review by egrefen — APPROVED** [review]
LGTM, new (and old) unit tests pass and are sensible. I guess for sanity check it would be good to check we can actually find the calls from the unit tests on our system, but I'm not sure how to do that myself.

**Comment by egrefen** [comment]
I'll let @mkozakov have a look rather than merge myself.

**Review by mkozakov — APPROVED** [review]
LGTM, do we also want to add this to the Node + GO sdks?

**Pull request merged**